docs/beefy.md

```diff
@@ -4,97 +4,364 @@ Related issues:
+     Best GRANDPA block───────────────────────────────┘        │
+                                                               │
+     Best produced block───────────────────────────────────────┘
+
+```
+
+A pseudo-algorithm of behavior for a fully-synced BEEFY validator is:
+
+```
+loop {
+  let (best_beefy_block, best_grandpa_block) = wait_for_best_blocks();
+
+  let block_to_vote_on = choose_next_beefy_block(
+    best_beefy_block,
+    best_grandpa_block
+  );
+
+  let payload_to_vote_on = retrieve_payload(block_to_vote_on);
+
+  let commitment = (block_to_vote_on, payload_to_vote_on);
+
+  let signature = sign_with_current_session_key(commitment);
+
+  broadcast_vote(commitment, signature);
+}
+```
+
+Read more about the details in [Implementation](#1-implementation) section.
+
+## Details
+
+Before we jump into describing how BEEFY works in details, let's agree on the terms we are going to
+use and actors in the system. All nodes in the network need to participate in the BEEFY networking
+protocol, but we can identify two distinct actors though: **regular nodes** and **BEEFY validators**.
+Validators are expected to actively participate in the protocol, by producing and broadcasting
+**votes**. Votes are simply their signatures over a **Commitment**. A Commitment consists of a
+**payload** (an opaque blob of bytes extracted from a block or state at that block, expected to be some
+form of crypto accumulator (like Merkle Tree Hash or Merkle Mountain Range Root Hash)) and **block
+number** from which this payload originates. Additionally Commitment contains BEEFY **validator
+set id** at that particular block. Note the block is finalized, so there is no ambiguity despite
+using block number instead of a hash. A collection of **votes**, or rather
+a Commitment and a collection of signatures is going to be called **Signed Commitment**. A valid
+(see later for the rules) Signed Commitment is also called a **BEEFY Justification** or
+**BEEFY Finality Proof**. For more details on the actual data structures please see
+[Data Formats](#2-data-formats).
+
+A **round** is an attempt by BEEFY validators to produce BEEFY Justification. **Round number** is
+simply defined as a block number the validators are voting for, or to be more precise, the
+Commitment for that block number. Round ends when the next round is started or when we receive ALL
+expected votes from ALL validators.
+
+Regular nodes are expected to:
+1. Receive & validate votes for the current round and broadcast them to their peers.
+1. Receive & validate BEEFY Justifications and broadcast them to their peers.
+1. Return BEEFY Justifications for [**Mandatory Blocks**](#3-round-selection) on demand.
+1. Optionally return BEEFY Justifications for non-mandatory blocks on demand.
+
+Validators are expected to additionally:
+1. Produce & broadcast vote for the current round.
+
+Both kinds of actors are expected to fully participate in the protocol ONLY IF they believe they
```

Comment:
So until they catch up they are fully passive?

Reply:
Correct. Added an extra sentence to emphasize that.